tests/scenarios/watch-mode-test.ts

```diff
@@ -263,37 +391,224 @@ app.forEachScenario(scenario => {
 
     hooks.beforeEach(async () => {
       app = await scenario.prepare();
-      cli = EmberCLI.launch(['serve', '--port', '0'], { cwd: app.dir });
-      await cli.ready;
-      cli.clearOutput();
+      server = EmberCLI.launch(['serve', '--port', '0'], { cwd: app.dir });
+      await waitFor(/Serving on http:\/\/localhost:[0-9]+\//, DEFAULT_TIMEOUT * 2);
+      server.clearOutput();
     });
 
     hooks.afterEach(async () => {
-      await cli.shutdown();
+      await server.shutdown();
     });
 
     test(`ember serve`, async function (assert) {
       const originalContent =
         'TWO IS A GREAT NUMBER< I LKE IT A LOT< IT IS THE POWER OF ALL  OF ELECTRONICS, MATH, ETC';
       assert.false(await checkScripts(/js$/, originalContent), 'file has not been created yet');
 
-      await fs.writeFile(path.join(app.dir, 'app/simple-file.js'), `export const two = "${originalContent}";`);
-      await waitFor('file added simple-file.js');
+      await appFile('app/simple-file.js').write(`export const two = "${originalContent}";`);
+      await added('simple-file.js');
       await waitFor(/Build successful/);
 
       assert.true(await checkScripts(/js$/, originalContent), 'the file now exists');
-      cli.clearOutput();
+      server.clearOutput();
 
       const updatedContent = 'THREE IS A GREAT NUMBER TWO';
       assert.false(await checkScripts(/js$/, updatedContent), 'file has not been created yet');
 
-      await fs.writeFile(path.join(app.dir, 'app/simple-file.js'), `export const two = "${updatedContent}";`);
-      await waitFor('file changed simple-file.js');
+      await appFile('app/simple-file.js').write(`export const two = "${updatedContent}";`);
+      await changed('simple-file.js');
       await waitFor(/Build successful/);
 
       // TODO: find a better way to test this; this seems to linger around
       // assert.false(await checkScripts(/js$/, originalContent), 'the original file does not exists');
       assert.true(await checkScripts(/js$/, updatedContent), 'the updated file now exists');
     });
+
+    Qmodule('[GH#1619] co-located components regressions', function (hooks) {
```

Comment:
I really like this style of test naming - it gives future viewers additional context as to why a test was added